plugins/coverage/info.js

```diff
@@ -19,10 +19,35 @@ module.exports = config => ({
             }
         },
         handler: (request, reply) => {
-            config.coveragePlugin
-                .getInfo(request.query)
-                .then(reply)
-                .catch(err => reply(boom.boomify(err)));
+            const { jobId, pipelineId, startTime, endTime, jobName, pipelineName, scope } = request.query;
+            const { jobFactory } = request.server.app;
+            const infoConfig = { jobId, pipelineId, startTime, endTime, jobName, pipelineName };
+
+            return Promise.resolve()
+                .then(() => {
+                    if (scope) {
```

Comment:
Thinking a bit more on this, do we even need all this ?

build.environment should have https://github.com/screwdriver-cd/coverage-sonar/pull/41/files#diff-168726dbe96b3ce427e7fedce31bb0bcR312 

This end point can simply take that value and avoid calling job model at all.

Only weirdness will be it's called an explicit Sonar Project Key, But that should be okay, in future if we add a new coverage source, we can handle it then.